### PR TITLE
[unicorn] update to 2.0.1.post1

### LINKED ISSUE #25707: [unicorn] update to 2.0.0
Library name:

New version number: 2.0.0

Other information that may be useful (release notes, etc...)
https://github.com/unicorn-engine/unicorn/releases

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #25707 

Update unicorn to the latest version 2.0.1.post1

Convert msbuild to cmake

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [ ] ~~SHA512s are updated for each updated download~~
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by wwiv** [comment]
This worked for WIN32, but didn't for linux. I had to make the following changes for it to build on Linux (Debian11).  Also the supports should  likely include "macos |" as well but I don't have that to test on.

```
rushfan@debian11:~/git/wwivdos/vcpkg$ git diff
diff --git a/ports/unicorn/portfile.cmake b/ports/unicorn/portfile.cmake
index 36d520036..0ad20c00e 100644
--- a/ports/unicorn/portfile.cmake
+++ b/ports/unicorn/portfile.cmake
@@ -2,12 +2,15 @@ if(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
     message(FATAL_ERROR "WindowsStore not supported")
 endif()
 
+
+if (WIN32)
 if(VCPKG_CRT_LINKAGE STREQUAL "dynamic" AND VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     message(FATAL_ERROR "unicorn can currently only be built with /MT or /MTd (static CRT linkage)")
 endif()
 
 # Note: this is safe because unicorn is a C library and takes steps to avoid memory allocate/free across the DLL boundary.
 set(VCPKG_CRT_LINKAGE "static")
+endif()  # WIN32
 
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
diff --git a/ports/unicorn/vcpkg.json b/ports/unicorn/vcpkg.json
index 51db5a76a..72979a8a1 100644
--- a/ports/unicorn/vcpkg.json
+++ b/ports/unicorn/vcpkg.json
@@ -4,7 +4,7 @@
   "description": "Unicorn is a lightweight multi-platform, multi-architecture CPU emulator framework",
   "homepage": "https://github.com/unicorn-engine/unicorn",
   "license": "GPL-2.0-only",
-  "supports": "windows & !uwp & !arm",
+  "supports": "linux | (windows & !uwp & !arm)",
   "dependencies": [
     {
       "name": "vcpkg-cmake",

```

**Comment by JonLiu1993** [comment]
@wwiv, Thanks for your testing and help, I will re-fix this pr locally and test your suggestions.

**Comment by cnSchwarzer** [comment]
Any update on this?

**Comment by closehandle** [comment]
Any update on this?

**Comment by BillyONeal** [comment]
@JonLiu is this OK to merge now?

**Comment by JonLiu1993** [comment]
Yes, I upgraded unicornD to the latest version `2.0.1.post1` https://github.com/unicorn-engine/unicorn/releases

File: ports/unicorn/fix-build.patch [hunk 1]
```diff
@@ -0,0 +1,22 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 4110ae5..a86b207 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -109,7 +109,7 @@ if(MSVC)
+     add_compile_options($<$<COMPILE_LANGUAGE:C>:/wd4267>)
+ 
+     # handle msvcrt setting being passed in CMAKE_C_FLAGS
+-    if(DEFINED CMAKE_MSVC_RUNTIME_LIBRARY)
++    if(NOT CMAKE_MSVC_RUNTIME_LIBRARY)
```
Comment (JonLiu1993) [inline ports/unicorn/fix-build.patch:10]: Fix error:
````
CMake Error at CMakeLists.txt:114 (message):
  please set msvcrt via CMAKE_C_FLAGS


-- Configuring incomplete, errors occurred!
````

File: ports/unicorn/fix-build.patch [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+         # do not support other methods of setting this (it would be more conformant, tho)
+         message(FATAL_ERROR "please set msvcrt via CMAKE_C_FLAGS")
+     endif()
+@@ -1431,7 +1431,7 @@ if(UNICORN_BUILD_TESTS)
+ endif()
+ 
+ 
+-if(UNICORN_INSTALL AND NOT MSVC)
++if(UNICORN_INSTALL)
```
Comment (JonLiu1993) [inline ports/unicorn/fix-build.patch:19]: Fix error:
````
ninja: error: unknown target 'install'
````

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**